### ISSUE TITLE
AliasSymbol.Equals throw NullReferenceException when comparing two global symbols

### ISSUE THREAD

**Issue by tcz717** [dotnet/roslyn#58999]
**Version Used**:  4.0.1

**Steps to Reproduce**:

1. Have two `global` `AliasSymbol` from two different documents, so that they are not **ReferenceEquals**.
2. Checking if they are equal by using `AliasSymbol.Equals` or `SymbolEqualityComparer.Default.Equals`

I am using ISymbol as the key of Dictionary. But when `global` symbol is added then get later, NullReferenceException is thrown. Here is the issue in my repo: https://github.com/tcz717/LsifDotnet/issues/8

I investigated and found that the `ContainingAssembly` property of the `global` symbol is null. But in the AliasSymbol.Equals code
https://github.com/dotnet/roslyn/blame/main/src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs#L254-L256, `ContainingAssembly` property is used without null checking. 

```csharp
            return (object?)other != null &&
                Equals(this.Locations.FirstOrDefault(), other.Locations.FirstOrDefault()) &&
                this.ContainingAssembly.Equals(other.ContainingAssembly, compareKind);
```

We should do null checking here or use `object.Equals`?

**Expected Behavior**:
The equal check should return `true`
**Actual Behavior**:
Throw NullReferenceException
```csharp
Unhandled exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.CodeAnalysis.CSharp.Symbols.AliasSymbol.Equals(Symbol obj, TypeCompareKind compareKind)
```

**Comment by jcouv** [comment]
Thanks. I'm able to repro now. The key was to use `GetAliasInfo` on `global` (I'd tried `GetSymbolInfo` which wouldn't return an alias).

**Comment by jcouv** [comment]
Fix merged for 17.2. Thanks for reporting the issue and the additional repro details.
